features/frameworks/greentea-client/greentea-client/test_env.h

```diff
@@ -28,6 +28,9 @@
 #endif
 
 #include <stdio.h>
+extern "C" {
```

Comment:
Unfortunately there is trade of to make, I don't believe we will get any thing very clean out of this. 

Splitting definitions into separate file does not reduce the number of `ifdef` in the code or if it does it is at the expense of user friendliness. 

First of all consider that if the proposed solution is chosen then declaration in `test_env_c.h` shall  be guarded with 

```c++ 
#ifdef __cplusplus
 extern "C" {
#endif

// declaration 

#ifdef __cplusplus
}
#endif
```
Otherwise if ever a C++ file include `test_env_c.h` the functions declared will be seen as `C++` symbols. Note that the inclusion guard in `test_env.h` can be removed.

Secondly, to allow inclusion of `test_env.h` within `C` code then `C++` bits have to be guarded. 

None of those guards are mandatory but they improve the end user experience.